pallets/file-system/src/utils.rs

```diff
@@ -1196,49 +1174,62 @@ where
         // Return the threshold needed for the BSP to be able to volunteer for the storage request.
         T::HashToThresholdType::convert(volunteering_hash)
     }
-}
 
-impl<T: crate::Config> shp_traits::SubscribeProvidersInterface for Pallet<T> {
-    type ProviderId = ProviderIdFor<T>;
+    pub fn compute_threshold_to_succeed(
+        bsp_id: &ProviderIdFor<T>,
+        requested_at: BlockNumberFor<T>,
+    ) -> Result<(T::ThresholdType, T::ThresholdType), DispatchError> {
+        let maximum_threshold: u32 = MaximumThreshold::<T>::get().into();
+        let global_weight = T::Providers::get_global_bsps_reputation_weight();
 
-    fn subscribe_bsp_sign_off(_who: &Self::ProviderId) -> DispatchResult {
-        // Adjust bsp assignment threshold by applying the inverse of the decay function after removing the asymptote
-        let mut bsp_assignment_threshold = BspsAssignmentThreshold::<T>::get();
-        let base_threshold =
-            bsp_assignment_threshold.saturating_sub(T::AssignmentThresholdAsymptote::get());
+        if global_weight
+            == <T::Providers as shp_traits::ReadProvidersInterface>::ReputationWeight::zero()
+        {
+            return Err(Error::<T>::NoGlobalReputationWeightSet.into());
+        }
 
-        bsp_assignment_threshold = base_threshold
-            .checked_div(&T::AssignmentThresholdDecayFactor::get())
-            .ok_or(Error::<T>::ThresholdArithmeticError)?
-            .saturating_add(T::AssignmentThresholdAsymptote::get());
+        // Global threshold starting point from which all BSPs begin their threshold slope. All BSPs start at this point
+        // with the starting reputation weight.
+        let threshold_global_starting_point = maximum_threshold
+            .checked_mul(ReplicationTarget::<T>::get().into() / global_weight.into() / 2)
```

Comment:
Maybe using some `saturated_div`s here? Also, this would saturate to zero in the divisions. `global_weight` is potentially a very large number, and `ReplicationTarget::<T>::get()` is something around 10. My suggestion would be to do:
```
maximum_threshold.saturated_div(global_weight).saturated_mul(ReplicationTarget::<T>::get()).saturated_div(2)
```
In that order. What you achieve is that:
1. In a regular scenario, `maximum_threshold` would be very large, and you'd start bringing it down with `global_weight`, also a large number. That way, when you multiply it by `ReplicationTarget::<T>::get()` you should still be within the numerical domain.
2. If `global_weight` is low still (the beginning of the network), when multiplying with `ReplicationTarget::<T>::get()` you'll get at most `u32::MAX` and then the threshold would be `u32::MAX / 2` (still pretty high).
3. If `maximum_threshold` is very low (like some times you set it in tests), the division would saturate to 1, and then the threshold would be `ReplicationTarget::<T>::get() / 2` (still very low).

Reply:
Given how much thought has to be put into this calculation and it's edge cases, probably worth adding some tests for it with different parameters.

Reply:
I have amended the formulas based on your comments. I had to use `checked_div` instead of `saturated_div` since it does not exist.